go/vt/vtgate/vindexes/vschema.go

```diff
@@ -165,7 +172,12 @@ func buildTables(source *vschemapb.SrvVSchema, vschema *VSchema) error {
 			default:
 				return fmt.Errorf("vindex %q needs to be Unique or NonUnique", vname)
 			}
-			vindexes[vname] = vindex
+			if _, ok := vschema.vindexes[vname]; ok {
```

Comment:
This approach seems ripe for confusion and seems fatally flawed to me.

Unless I'm missing something, this will put the vindex in the "unique" map if there is an _odd_ number of vindexes with the same name, not if there is just one. For example when there are three, the first match puts it in the hash, the second takes it out, and the third puts it back in again.

Also, from a naming standpoint,  having `VSchema.vindexes` be the map of globally-unique ones is not at all obvious to someone coming in fresh to this code. Assuming we keep this approach at all (see my other comment), I'd suggest changing the field name to `uniqueVindexes` or something more obvious. I also think we should change`VSchema.tables` at the same time since it has the same confusing naming semantics and the same issue.

Reply:
This is a go-ism: A map entry can exist as `nil`. The `ok` check tells you if the entry exists or not, even if it's `nil`:
* `!ok`: vindex does not exist.
* `nil`: multiple vindexes of the same name exist.
* `!= nil`: there is only one vindex with this name.

I agree with the naming. I'll add a `unique` prefix to both this and the tables variable.

Reply:
Ah of course... I've been buried in php land the last few days where there's no difference between nonexistent and null. This makes sense and restores my momentarily-ever-so-slightly-in-doubt faith in the code correctness :)